src/transport/SessionManager.cpp

```diff
@@ -773,29 +786,59 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
         return;
     }
 
-    System::PacketBufferHandle msgCopy;
+    // Extract MIC from the end of the message.
+    uint8_t * data     = msg->Start();
+    uint16_t len       = msg->DataLength();
+    uint16_t footerLen = packetHeader.MICTagLength();
+    VerifyOrReturn(footerLen <= len);
+
+    uint16_t taglen = 0;
+    MessageAuthenticationCode mac;
+    ReturnOnFailure(mac.Decode(packetHeader, &data[len - footerLen], footerLen, &taglen));
+    VerifyOrReturn(taglen == footerLen);
+
     bool decrypted = false;
     while (!decrypted && iter->Next(groupContext))
     {
+        CryptoContext context(groupContext.keyContext);
+        msgCopy = msg.CloneData();
```

Comment:
This is where the `msgCopy.IsNull()` check needs to be, no?  Before we start dereferencing it...

Reply:
Added extra IsNull check after clone.